src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -239,41 +235,26 @@ private void flushData(final EventTypePartition pk, final List<ConsumedEvent> da
                            final Optional<String> metadata) {
         try {
             final NakadiCursor sentOffset = offsets.get(pk).getSentOffset();
-            final String batch = serializeBatch(sentOffset, data, metadata);
+            final SubscriptionCursor cursor = getContext().getCursorConverter().convert(
+                    sentOffset,
+                    getContext().getCursorTokenService().generateToken());
+
+            final int batchSize = getContext().getWriter().writeSubscriptionBatch(
+                    getOut().getOutputStream(),
+                    cursor,
+                    data,
+                    metadata);
+
+            getOut().getOutputStream().flush();
```

Comment:
In my opinion it is the work of the writer, because it writes data to the output stream, and we never know when the system will flush its buffers. From other hand work with output stream in different parts of the code dilutes the usage of the output stream which is not good for the future work with the code.